packages/tendermint-rpc/src/tendermint34/adaptor/responses.ts

```diff
@@ -743,8 +743,8 @@ interface RpcBlock {
   };
   // It's currently unclear why the deep nesting is requied.
   // See https://github.com/tendermint/tendermint/issues/7697.
-  readonly evidence?: {
-    readonly evidence?: readonly RpcEvidence[];
+  readonly evidence: {
+    readonly evidence: readonly RpcEvidence[];
```

Comment:
Ah, this should remain unchanged. The private type `RpcBlock` is what we expect from the Tendermint API. Array encoding coming from Go can be strange and unexpected. Empty arrays are often omitted.

Better turn undefined to an empty array in this code:

```ts
evidence: data.evidence?.evidence,
```